### PR TITLE
Add conversions from bytes

### PR DESCRIPTION
**Pull request by andreiltd**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -278,6 +280,64 @@ impl From<Ulid> for u128 {
...
+            | u128::from(bytes[8]) << 56
+            | u128::from(bytes[9]) << 48
+            | u128::from(bytes[10]) << 40
+            | u128::from(bytes[11]) << 32
+            | u128::from(bytes[12]) << 24
+            | u128::from(bytes[13]) << 16
+            | u128::from(bytes[14]) << 8
+            | u128::from(bytes[15]);
+
+        Self(value)
```
Comment (dylanhart) [inline src/lib.rs:302]: Prefer existing conversion functions:

`Self(u128::from_be_bytes(bytes))`

File: src/lib.rs [hunk 2]
```diff
@@ -278,6 +280,64 @@ impl From<Ulid> for u128 {
...
+    }
+}
+
+impl From<Ulid> for [u8; 16] {
+    fn from(ulid: Ulid) -> Self {
+        let value = ulid.0;
+
+        [
```
Comment (dylanhart) [inline src/lib.rs:310]: Prefer existing conversion functions:

`self.0.to_be_bytes()`

File: src/lib.rs [hunk 3]
```diff
@@ -278,6 +280,64 @@ impl From<Ulid> for u128 {
...
+            ((value >> 32) & 0xFF) as u8,
+            ((value >> 24) & 0xFF) as u8,
+            ((value >> 16) & 0xFF) as u8,
+            ((value >> 8) & 0xFF) as u8,
+            (value & 0xFF) as u8,
+        ]
+    }
+}
+
+impl TryFrom<&[u8]> for Ulid {
```
Comment (dylanhart) [inline src/lib.rs:331]: Remove in favor of letting the caller assert the length. We can add docs similar to <https://doc.rust-lang.org/nightly/std/primitive.u128.html#method.from_be_bytes>.

File: src/lib.rs [hunk 4]
```diff
@@ -278,6 +280,64 @@ impl From<Ulid> for u128 {
     }
 }
 
+impl From<[u8; 16]> for Ulid {
```
Comment (dylanhart) [inline src/lib.rs:283]: Can add a `const fn Ulid::from_bytes(_: [u8; 16])` function to allow this to be used in a const context.

File: src/lib.rs [hunk 5]
```diff
@@ -278,6 +280,64 @@ impl From<Ulid> for u128 {
...
+            | u128::from(bytes[12]) << 24
+            | u128::from(bytes[13]) << 16
+            | u128::from(bytes[14]) << 8
+            | u128::from(bytes[15]);
+
+        Self(value)
+    }
+}
+
+impl From<Ulid> for [u8; 16] {
```
Comment (dylanhart) [inline src/lib.rs:306]: Can add a `const fn Ulid::to_bytes(): [u8; 16]` function to allow this to be used in a const context.

**Review by dylanhart — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 2 above)
Comment (andreiltd) [reply src/lib.rs]: :heavy_check_mark:

File: src/lib.rs (hunk 1 above)
Comment (andreiltd) [reply src/lib.rs]: Done!

File: src/lib.rs (hunk 4 above)
Comment (andreiltd) [reply src/lib.rs:314]: Done!

File: src/lib.rs (hunk 5 above)
Comment (andreiltd) [reply src/lib.rs:320]: Done!

File: src/lib.rs (hunk 3 above)
Comment (andreiltd) [reply src/lib.rs]: Removed, where do we want the docs for this one,  `from_bytes` ?

Comment (dylanhart) [reply src/lib.rs]: That would be a good spot, going to merge as-is though.

**Review by dylanhart — APPROVED** [review]

**Pull request merged**